apache-solr/README.md

```diff
@@ -0,0 +1,73 @@
+# Apache Solr and ZooKeeper Cluster Sample App
+
+This project sets up a cluster of Apache Solr and ZooKeeper instances, integrated with Grafana agent for metric and log collection. It uses Multipass to create virtual machines where Apache Solr and ZooKeeper are deployed and configured. Additionally, a Grafana Agent is installed for metrics and logs collection and forwarding.
+
+## Prerequisites
+
+- Multipass: Used to create and manage VMs.
+- Grafana Cloud Account: For metrics and log forwarding (optional).
+- Basic understanding of Solr, ZooKeeper, and Grafana.
+
+## Versions
+Apache Solr: 8.11.2
+ZooKeeper: 3.4.13
+Grafana Agent: 0.39.0-1 for arm64 architecture (adjust as per your architecture)
+
+## Configuration
+
+Before running the setup, ensure to fill out the `grafana-agent-template.yaml` with your Grafana Cloud credentials and endpoints. Replace placeholders like `<your grafana.com password>`, `<your grafana.com username>`, `<your prometheus push endpoint>`, etc., with actual values. Rename the file to `grafana-agent.yaml` after configuration.
```

Comment:
This works fine for an interactive user doing something like acceptance testing, but makes it difficult to automate, which is an ultimate goal.

We haven't defined a repeatable/shared way to do this yet (some shared libs/scripts/patterns are forthcoming), but in all of the sample apps I've written, I'm using jinja templates and a Docker container execution to render those templates with appropriate variables. I.E. You can ask the user on the shell for the variables, then feed them into jinja templates.

I'm open to other/better solutions, and I would like to make those steps simpler/reusable in this repo.

Example from the MySQL Sample App: https://github.com/grafana/integration-sample-apps/blob/c0916d21dae6b9d130af99e00226e0954609b450/mysql/Makefile#L13-L41

Reply:
Added jinja templates for the grafana agent section